server/src/main/java/org/elasticsearch/indices/recovery/plan/SnapshotsRecoveryPlannerService.java

```diff
@@ -126,6 +125,16 @@ private ShardRecoveryPlan computeRecoveryPlanWithSnapshots(@Nullable String shar
         );
     }
 
+    private boolean isSnapshotVersionCompatible(ShardSnapshot snapshot) {
```

Comment:
Can we add a comment that this runs on the primary, but using it's version is fine, since `NodeVersionAllocationDecider` ensures we only recover to a node that has newer or same version.